dask/dataframe/dask_expr/_expr.py

```diff
@@ -3804,19 +3806,26 @@ def __le__(self, other):
         return True
 
 
-def determine_column_projection(expr, parent, dependents, additional_columns=None):
+def determine_column_projection(
+    expr: Expr,
+    parent: Expr,
+    dependents: dict[str, Collection[weakref.ref[BaseExpr]]],
+    additional_columns: list | None = None,
+):
     if isinstance(parent, Index):
         column_union = []
     else:
         column_union = parent.columns.copy()
-    parents = [x() for x in dependents[expr._name] if x() is not None]
+    parents: list[Expr]
+    parents = [
+        x() for x in dependents[expr._name] if isinstance(x(), Expr)  # type: ignore
```

Comment:
Is there a specific code to ignore here?

Reply:
without evaluating the expr twice this can go